File: src/nl/rubensten/texifyidea/structure/LatexFilePresentation.java
package nl.rubensten.texifyidea.structure;

import com.intellij.navigation.ItemPresentation;
import com.intellij.psi.PsiFile;
import nl.rubensten.texifyidea.TexifyIcons;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

/**
 * @author Ruben Schellekens
 */
public class LatexFilePresentation implements ItemPresentation {

    private final PsiFile file;

    public LatexFilePresentation(PsiFile file) {
        this.file = file;
    }

    @Nullable
    @Override
    public String getPresentableText() {
        return file.getName();
    }

    @Nullable
    @Override
    public String getLocationString() {
        return file.getVirtualFile().getPath();
    }

    @Nullable
    @Override
    public Icon getIcon(boolean b) {
        return TexifyIcons.DOT_LATEX;
    }
}

File: src/nl/hannahsten/texifyidea/structure/latex/LatexFilePresentation.kt
package nl.hannahsten.texifyidea.structure.latex

import com.intellij.navigation.ItemPresentation
import com.intellij.psi.PsiFile
import nl.hannahsten.texifyidea.TexifyIcons
import nl.hannahsten.texifyidea.util.magic.PatternMagic

/**
 * @author Hannah Schellekens
 */
open class LatexFilePresentation(private val file: PsiFile) : ItemPresentation {

    private val presentableText = PatternMagic.fileExtension.matcher(file.name).replaceAll("")

    override fun getPresentableText() = presentableText!!

    override fun getLocationString() = file.virtualFile.path

    override fun getIcon(b: Boolean) = TexifyIcons.DOT_LATEX
}